src/secrets/CommandEdit.ts

```diff
@@ -54,41 +55,85 @@ class CommandEdit extends CommandPolykey {
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = response.secretContent;
+
         const tmpDir = await fs.promises.mkdtemp(
           path.join(os.tmpdir(), 'polykey-'),
         );
         const tmpFile = path.join(tmpDir, 'pksecret');
-        await this.fs.promises.writeFile(tmpFile, secretContent);
-        execSync(`$EDITOR \"${tmpFile}\"`, { stdio: 'inherit' });
-        let content: Buffer;
+        const secretExists = await binUtils.retryAuthentication(
+          async (auth) => {
+            let exists: boolean = true;
+            try {
+              const response =
+                await pkClient.rpcClient.methods.vaultsSecretsGet({
+                  metadata: auth,
+                  nameOrId: secretPath[0],
+                  secretName: secretPath[1],
+                });
+              await this.fs.promises.writeFile(tmpFile, response.secretContent);
+            } catch (e) {
+              const [cause, _] = binUtils.remoteErrorCause(e);
+              if (cause instanceof vaultsErrors.ErrorSecretsSecretUndefined) {
+                exists = false;
+              } else {
+                throw e;
+              }
+            }
+            return exists;
+          },
+          meta,
+        );
+        // If the editor exited with a code other than zero, then execSync
+        // will throw an error. So, in the case of saving the file but the
+        // editor crashing, the program won't save the updated secret.
+        execSync(`${process.env.EDITOR} \"${tmpFile}\"`, { stdio: 'inherit' });
+        let content: string;
         try {
-          content = await this.fs.promises.readFile(tmpFile);
+          content = (await this.fs.promises.readFile(tmpFile)).toString(
+            'binary',
+          );
         } catch (e) {
-          throw new errors.ErrorPolykeyCLIFileRead(e.message, {
-            data: {
-              errno: e.errno,
-              syscall: e.syscall,
-              code: e.code,
-              path: e.path,
-            },
-            cause: e,
-          });
+          // If the secret exists but the file doesn't, then something went
+          // wrong, and the file cannot be read anymore. This is bad.
+          // If the secret didn't exist before and we can't read the file,
+          // then the secret was never actually created or saved. The user
+          // doesn't want to make the secret anymore, so abort mision!
+          if (e.code === 'ENOENT' && secretExists) {
+            throw new errors.ErrorPolykeyCLIFileRead(e.message, {
+              data: {
+                errno: e.errno,
+                syscall: e.syscall,
+                code: e.code,
+                path: e.path,
+              },
+              cause: e,
+            });
+          }
+          return;
         }
```

Comment:
This is bad, we can't assume we'll never get any other error. The logic seems to be, if `ENOENT` and it existed then we want an error, if `ENOENT` and file didn't exist then we just allow it. but right now all other errors are ignored and that is really bad.

The 3rd case of other errors happening must be re-thrown. This is important.